kernel/src/hil/hmac.rs

```diff
@@ -0,0 +1,39 @@
+//! Interface for HMAC
+
+use crate::returncode::ReturnCode;
+
+/// Implement this trait and use `set_client()` in order to receive callbacks.
+pub trait Client<'a> {
+    // On success the `result` will contain the digest. On error it will contain
+    // the HMAC error value.
+    fn hash_done(&'a self, result: Result<&'a mut [u8], u32>);
+}
+
+pub const HMAC_KEY_SIZE: usize = 16;
```

Comment:
AFAIK it is recommended that the key size is the same block size of the hash function.
This naming seems a bit ambiguous but I guess it's directly related `lowrisc/hmac`?

Can we make this an associated const in `Hmac trait` to be more abstract and to work with different key sizes and hash algorithms?

Can we make it an associated type in the `Hmac` trait instead?

Reply:
Yep, I have updated this to add a function that sets the hash algorithm and key size.